cmd/dfdaemon/app/root.go

```diff
@@ -41,45 +43,139 @@ var rootCmd = &cobra.Command{
 	Long:              "The dfdaemon is a proxy between container engine and registry used for pulling images.",
 	DisableAutoGenTag: true, // disable displaying auto generation tag in cli docs
 	RunE: func(cmd *cobra.Command, args []string) error {
-		initOption(opt)
+		if viper.GetBool("version") {
+			fmt.Println("dfdaemon version:", version.DFDaemonVersion)
+			return nil
+		}
+
+		if err := readConfigFile(viper.GetViper(), cmd); err != nil {
+			return errors.Wrap(err, "read config file")
+		}
 
-		s, err := dfdaemon.NewFromConfig(*g.Properties, *opt)
+		cfg, err := getConfigFromViper(viper.GetViper())
 		if err != nil {
-			logrus.Fatal(err)
-			return err
+			return errors.Wrap(err, "get config from viper")
+		}
+
+		if viper.GetBool("verbose") {
+			yaml.NewEncoder(os.Stdout).Encode(cfg)
+		}
+
+		if err := initDfdaemon(*cfg); err != nil {
+			return errors.Wrap(err, "init dfdaemon")
 		}
 
+		s, err := dfdaemon.NewFromConfig(*cfg)
+		if err != nil {
+			return errors.Wrap(err, "create dfdaeon from config")
+		}
 		return s.Start()
 	},
 }
 
 func init() {
-	opt.AddFlags(rootCmd.Flags())
+	self, err := filepath.Abs(os.Args[0])
```

Comment:
exec.LookPath(os.Args[0])?

Reply:
Oh, I forgot the case where we call dfdaemon with only `dfdaemon` which searches through `PATH`. `exec.LookPath` should be used indeed.

Reply:
Fixed